tests/unit_tests/gui/test_main_window.py

```diff
@@ -41,9 +43,19 @@
 from ert.shared.models import EnsembleExperiment, MultipleDataAssimilation
 
 
+def find_cases_dialog_and_panel(
+    gui: ErtMainWindow, qtbot: QtBot
+) -> Tuple[ClosableDialog, CaseInitializationConfigurationPanel]:
+    qtbot.waitUntil(lambda: gui.findChild(ClosableDialog, "manage-cases") is not None)
+    dialog = gui.findChild(ClosableDialog, "manage-cases")
+    cases_panel = dialog.findChild(CaseInitializationConfigurationPanel)
+    assert isinstance(cases_panel, CaseInitializationConfigurationPanel)
```

Comment:
Have you tested what is the expected behaviour if that assertion fails?

Reply:
the assert does two things, it checks that you have found what you were looking for (otherwise you get `None` which is not a `CaseInitializationConfigurationPanel`) and it is a typeguard which helps type checkers figure out what cases_panel is.